Code/GraphMol/Descriptors/AUTOCORR3D.cpp

```diff
@@ -0,0 +1,242 @@
+//
+//  Copyright (c) 2012, Institue of Cancer Research.
+//  All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+//modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above
+//       copyright notice, this list of conditions and the following
+//       disclaimer in the documentation and/or other materials provided
+//       with the distribution.
+//     * Neither the name of Institue of Cancer Research.
+//       nor the names of its contributors may be used to endorse or promote
+//       products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Guillaume GODIN access the AutoCorrelation 3D descriptors names in Dragon TDB
+
+
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/MolTransforms/MolTransforms.h>
+
+#include "AUTOCORR3D.h"
+#include "MolData3Ddescriptors.h"
+
+#include "GraphMol/PartialCharges/GasteigerCharges.h"
+#include "GraphMol/PartialCharges/GasteigerParams.h"
+#include <Numerics/EigenSolvers/PowerEigenSolver.h>
+
+#include <Numerics/Matrix.h>
+#include <Numerics/SquareMatrix.h>
+#include <Numerics/SymmMatrix.h>
+#include <boost/foreach.hpp>
+#include <math.h>
+#include <Eigen/Dense>
+#include <Eigen/SVD>
+#include <iostream>
+#include <Eigen/Core>
+#include <Eigen/QR>
+
+
+
+using namespace Eigen;
+namespace RDKit {
+namespace Descriptors{
+
+namespace {
+
+MolData3Ddescriptors moldata3D;
+
+double* retreiveMat(MatrixXd matrix) {
```

Comment:
do you have a better proposal for the naming ? this subfunction will only convert Eigen mat to C++ normal double pointer.